dice-game-ui-angular/src/app/services/user-service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from '@angular/core';
+import { Subject, Observable } from 'rxjs';
+import { SocialAuthService, GoogleLoginProvider, FacebookLoginProvider } from 'angularx-social-login';
+
+export type AuthUserEventHandler = (user: User) => void;
+
+export interface User {
+  id: string;
+  name: string;
+  photoUrl: string;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class UserService {
+  private readonly localUser: User;
+  private user: User;
+  private userSubject = new Subject<User>();
+
+  constructor(private authService: SocialAuthService) {
+    this.localUser = { id: 'local player', name: 'local player' } as User;
+    this.user = this.localUser;
+
+    this.authService.authState.subscribe((socialUser) => {
+      this.user = socialUser
```

Comment:
I suppose, using of current user variable could be avoided by getting rid of `getCurrentUserId` and `isLoggedIn` methods, or rewriting them to observables.
If you still need to store current user locally, you can use `BehaviorSubject` instead of simple `Subject` and get current value from it.

Reply:
Fixed